### PR TITLE
#278 Improve navigation menu

### PR DESCRIPTION
**Pull request by GeorgiChochov**
1. Split Master and Detail Pages per entity.
2. Extracted add/edit functionality to separate `.js` files per entity to avoid duplication in `Master-Page controller (add)` and `Detail-Page controller (edit)`. This required updating the controller declarations from the deprecated declaration syntax: 
```
sap.ui.controller("controllerName", {
})
```

to the current: 

```
sap.ui.define([
  "sap/ui/core/mvc/Controller"
], function (Controller) {
  return Controller.extend("components.dataset.datasetMain", {
  })
});
```

The new declaration requires one extra level of indentation, so the changes look broader than they actually are. Mostly whitespace outside fo the `init` function and the functions related to the `add/edit` functionality.

### CODE REVIEW

**Comment by Zejnilovic** [comment]
Pre-selects in the menu make it so, that I am not able to choose the first item and have to first choose a different one. This is not true for `Run`, they have nothing preselected and anything can be chosen.

Apr-02-2019 12-46-54

File: menas/ui/components/dataset/datasetMain.controller.js [hunk 1]
```diff
@@ -12,320 +12,225 @@
...
+      let current = this._model.getProperty("/currentDataset");
 
-    sap.m.MessageBox.show("This action will remove all versions of the dataset definition. \nAre you sure?.", {
-      icon: sap.m.MessageBox.Icon.WARNING,
-      title: "Are you sure?",
-      actions: [sap.m.MessageBox.Action.YES, sap.m.MessageBox.Action.NO],
-      onClose: function (oAction) {
-        if (oAction === "YES") {
-          DatasetService.disableDataset(current.name)
+      sap.m.MessageBox.show("This action will remove all versions of the dataset definition. \nAre you sure?.", {
```
Comment (Zejnilovic) [inline menas/ui/components/dataset/datasetMain.controller.js]: There is a dot `.` after the question mark `?`

File: menas/ui/components/run/runMaster.view.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  -->
+
+<core:View id="runMasterView" xmlns:core="sap.ui.core" xmlns="sap.m"
+           controllerName="components.run.runMaster"
+           xmlns:cust="http://schemas.sap.com/sapui5/extension/sap.ui.core.CustomData/1">
+    <Page id="masterPage" title="Runs" showNavButton="true" navButtonPress="onPressMasterBack">
+        <content>
+            <List items="{runs>/}" mode="SingleSelectMaster" selectionChange="runSelected"
+                  noDataText="No Runs">
+                <StandardListItem title="{runs>datasetName}(v{runs>datasetVersion}) - Run {runs>runId}"
```
Comment (Zejnilovic) [inline menas/ui/components/run/runMaster.view.xml]: Some places we have `(v.XX)` and some places `(vXX)`. This should maybe be unified.

File: menas/ui/components/mappingTable/mappingTableMain.controller.js [hunk 3]
```diff
@@ -13,415 +13,328 @@
...
+      let schemas = this._model.getProperty("/schemas");
+      if (!schemas || schemas.length === 0) {
+        SchemaService.getSchemaList(false, true);
+      }
+    },
+
+    onRemovePress: function (oEv) {
+      let current = this._model.getProperty("/currentMappingTable")
+
+      MessageBox.show("This action will remove ALL versions of the mapping table definition. \nAre you sure?.", {
```
Comment (Zejnilovic) [inline menas/ui/components/mappingTable/mappingTableMain.controller.js]: Same as before `dot` after `question mark`
Second thing is, why do we have all caps `ALL` here and not up there.

File: menas/ui/components/schema/schemaMaster.controller.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      this._router.navTo("schemas", {
+        id: selected
+      });
+    },
+
+    onAddPress: function () {
+      let oView = this.getView();
+
+      let addSchemaDialog = this.byId("addSchemaDialog");
+      if (!this.byId("addSchemaDialog")) {
```
Comment (Zejnilovic) [inline menas/ui/components/schema/schemaMaster.controller.js]: switch the if else, so there is no needles negation and the code might be easier to understand.

File: menas/ui/service/RunService.js [hunk 5]
```diff
@@ -17,18 +17,28 @@ jQuery.sap.require("sap.m.MessageBox");
...
+        if (oData.length > 0 && oControl) {
           let firstDataset = oData[0];
-          this.getRun(oDetailPage, oTable, firstDataset.datasetName, firstDataset.datasetVersion, firstDataset.runId)
+          this.getRun(oControl, oTable, firstDataset.datasetName, firstDataset.datasetVersion, firstDataset.runId)
         }
       },
       () => {
         sap.m.MessageBox
-          .error("Failed to get the list of runs. Please wait a moment and try reloading the application")
+          .error("Failed to get the any run. Please wait a moment and try reloading the application")
```
Comment (Zejnilovic) [inline menas/ui/service/RunService.js]: remove `the` in `the any run`

**Comment by Zejnilovic** [comment]
**Idea proposal**: Make the menu visible nonstop on widths more than XXX (e.g. 1280). Otherwise, make it open using the hamburger menu.

**Comment by GeorgiChochov** [comment]
> **Idea proposal**: Make the menu visible nonstop on widths more than XXX (e.g. 1280). Otherwise, make it open using the hamburger menu.

That makes sense, maybe create a task for it.

**Comment by Zejnilovic** [comment]
> > **Idea proposal**: Make the menu visible nonstop on widths more than XXX (e.g. 1280). Otherwise, make it open using the hamburger menu.
> 
> That makes sense, maybe create a task for it.

Sure. I will create one. What about the pre-selects?

**Comment by GeorgiChochov** [comment]
> Sure. I will create one. What about the pre-selects?

Working on it, as well as the navigation to new entity on creation.

**Comment by Zejnilovic** [comment]
> > **Idea proposal**: Make the menu visible nonstop on widths more than XXX (e.g. 1280). Otherwise, make it open using the hamburger menu.
> 
> That makes sense, maybe create a task for it.

Task created #281

**Pull request merged**